Automattic/wp-calypso
# Signup: Refactor Domain Origin Logic To Use Dependency Store

Fixes 113-gh-Automattic/growth-foundations

### Summary
The logic for tracking the domain origin prop for the `calypso_signup_complete` event was implemented in this PR: [Signup: Add Domain Origin Prop to Tracks Event](https://github.com/Automattic/wp-calypso/pull/78085) which wasn't following best code practices because it used `localStorage` for tracking the data. This PR utilizes the dependency store of the signup framework to track that data across the steps.

### Testing Instructions
1. Checkout this branch
2. Run `yarn` and `yarn start`
3. Go to `http://calypso.localhost:3000/start`
4. Open [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) so you can monitor tracks events
5. Click on **Choose my domain later** in the sidebar. Continue through the flow until you see the `calypso_signup_complete` event fire in tracks vigilante. Confirm the even has the following property:
`signup_domain_origin: 'choose-later'`
6. Click on **Use a domain I own** in the sidebar. Continue through the flow until you see the `calypso_signup_complete` event fire in tracks vigilante. Confirm the even has the following property:
`signup_domain_origin: 'use-your-domain'`
7. Select the free .wordpress.com domain. Continue through the flow until you see the `calypso_signup_complete` event fire in tracks vigilante. Confirm the even has the following property:
`signup_domain_origin: 'free'`
8. Select the one of the custom suggested domains. Continue through the flow until you see the `calypso_signup_complete` event fire in tracks vigilante. Confirm the even has the following property:
`signup_domain_origin: 'custom'`
9. If for some reason the domain origin hasn't been set, the value of the `signup_domain_origin` should be `not-set`
